src/main/java/io/vertx/core/json/JsonObject.java

```diff
@@ -70,6 +70,32 @@ public JsonObject(Map<String, Object> map) {
   }
 
   /**
+   * Create a JsonObject from the fields of a Java object.
+   * Faster than calling `new JsonObject(Json.encode(pojo))`.
+   * 
+   * @param obj
+   *          The object to convert to a JsonObject.
+   * @throws IllegalArgumentException
+   *          if conversion fails due to an incompatible type.
+   */
+  @SuppressWarnings("unchecked")
+  public JsonObject(Object obj) {
```

Comment:
@lukehutch agreed, however it should be renamed with a name like `mapTo` / `mapFrom` in both case that has the semantic that mappings happens.